test/ctia/http/generative/fulltext_search_test.clj

```diff
@@ -360,3 +362,40 @@
              (-> res first
                  (select-keys (keys expected))
                  (= expected))))))))))
+
+(deftest enforcing-fields-with-feature-flag-test
+  (testing "feature flag set? fields should be enforced"
+    (are [properties query expected-fields]
+         (helpers/with-properties properties
+           (helpers/fixture-ctia-with-app
```

Comment:
this starts CTIA each time. Any alternative?
The feature flag service is tested, then I would unit test `enforce-search-fields` (even if its code is trivial), and keep only one test case here to check that default search fields are properly disabled: `["ctia.feature-flags" "enforce-search-fields:false"]`. The 2 other test cases should be covered with the all the other search tests.